jax/interpreters/numpy_eval.py

```diff
@@ -0,0 +1,67 @@
+# Copyright 2018 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+from contextlib import contextmanager
+from typing import Any, Dict, Callable
+
+from jax import config, core
+
+np_impl: Dict[core.Primitive, Callable[..., Any]] = {}
+
+def is_tracing():
+  if not config.omnistaging_enabled:
+    return False
+  # TODO(mattjj) improve:
+  trace = core.thread_local_state.trace_state.trace_stack.stack[0]
+  return trace.trace_type is NumpyEvalTrace
+
+class NumpyEvalTrace(core.Trace):
+  def pure(self, x): return x
+  lift = sublift = pure
+
+  def process_primitive(self, primitive, tracers, params):
+    impl = np_impl.get(primitive)
+    if impl is None:
+      raise NotImplementedError(f"No original numpy implementation for "
+                                f"'{primitive}' primitive found.")
+    return impl(*tracers, **params)
+
+  def process_call(self, primitive, f, tracers, params):
+    return f.call_wrapped(*tracers)
+  process_map = process_call
+
+@contextmanager
+def numpy_eval():
+  """
+  Makes JAX code evaluate with NumPy instead of XLA.
+
+  >>> @numpy_eval()
+  ... def some_fun(...):
+  ...  # jax code
+
+  Inline:
+
+  >>> from jax.interpreters.numpy_eval import numpy_eval
+  ... with numpy_eval():
+  ...   # jax code
+
+  `jit(numpy_eval()(some_fun))` will collapse constant calculations using NumPy,
```

Comment:
BTW, this is restructured text, not markdown, so I think you need double backticks.

(Although you haven't added this to the public docs, so I guess it's moot because it won't be rendered anyway...)